### PR TITLE
#81 remove dummy workflows and upload release artifacts

### PR DESCRIPTION
**Pull request by tomuben**

### CODE REVIEW

File: doc/changes/changes_7.1.0.md [hunk 1]
```diff
@@ -7,7 +7,8 @@ Code name: t.b.d.
 t.b.d.
 
 ## Features / Enhancements
-n/a
+
+ - #81: Remove dummy workflows and upload release artifacts
```
Comment (morazow) [inline doc/changes/changes_7.1.0.md:11]:
```suggestion
 - #81: Removed dummy workflows and added upload release artifacts workflows
```

File: tests/run_test_within_docker.sh [hunk 2]
```diff
@@ -22,7 +22,7 @@ if [ $# -gt 1 ]; then
   case "$2" in
     valgrind)
       export DEBIAN_FRONTEND=noninteractive
-      apt install -y valgrind
+      apt-get update && apt install -y valgrind
```
Comment (morazow) [inline tests/run_test_within_docker.sh:25]: Should we put `-y` after update also?

File: .github/workflows/release_droid_upload_github_release_assets.yml [hunk 3]
```diff
@@ -8,8 +8,126 @@ on:
...
+
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          r-version: ${{ matrix.config.r }}
+
+      - name: Install openssl MacOsX
+        if: matrix.config.os == 'macOS-latest'
+        run: |
+          brew install openssl
+          BREW_PREFIX=`brew --prefix`
```
Comment (morazow) [inline .github/workflows/release_droid_upload_github_release_assets.yml:32]: Do we need `BREW_PREFIX` env variable?

File: .github/workflows/release_droid_upload_github_release_assets.yml [hunk 4]
```diff
@@ -8,8 +8,126 @@ on:
...
+      - name: Install Ubuntu package dependencies
+        if: matrix.config.os == 'ubuntu-latest'
+        run: |
+            sudo apt-get -y update && \
+               sudo apt-get -y install --no-install-recommends libcurl4-openssl-dev curl libxml2-dev  libssl-dev && \
```
Comment (morazow) [inline .github/workflows/release_droid_upload_github_release_assets.yml:37]:
```suggestion
               sudo apt-get -y install --no-install-recommends libcurl4-openssl-dev curl libxml2-dev libssl-dev && \
```

File: .github/workflows/release_droid_upload_github_release_assets.yml [hunk 5]
```diff
@@ -8,8 +8,126 @@ on:
...
+            mkdir -p "${out_put_dir}"
+            echo "output_dir=${out_put_dir}" >> $GITHUB_ENV
+        shell: bash
+      - name: Build
+        run: |
+          devtools::build(binary=TRUE, path='${{ env.output_dir }}')
+        shell: Rscript {0}
+        env: # Set individual GH PAT
+          GITHUB_PAT: ${{ secrets.PAT_FOR_DEVTOOLS }}
+      - name: Check result
```
Comment (morazow) [inline .github/workflows/release_droid_upload_github_release_assets.yml:64]: Should this have `ubuntu-latest` condition? Without it, it seems to be run for all distros.

File: .github/workflows/release_droid_upload_github_release_assets.yml [hunk 6]
```diff
@@ -8,8 +8,126 @@ on:
...
+        run: |
+           echo "Target folder is ${{ env.output_dir }}"
+           ls "${{ env.output_dir }}"
+      - name: Upload assets to the GitHub release draft
+        uses: shogo82148/actions-upload-release-asset@v1
+        with:
+          upload_url: ${{ github.event.inputs.upload_url }}
+          asset_path: ${{ runner.temp }}/bin/source/*
```
Comment (morazow) [inline .github/workflows/release_droid_upload_github_release_assets.yml:133]: We also try to upload sha256 sums of artifacts, maybe good idea also to upload them. Example from java: https://github.com/exasol/hive-virtual-schema/blob/main/.github/workflows/release_droid_upload_github_release_assets.yml#L32-L43

**Review by morazow — CHANGES_REQUESTED** [review]

File: .github/workflows/release_droid_upload_github_release_assets.yml (hunk 6 above)
Comment (morazow) [reply .github/workflows/release_droid_upload_github_release_assets.yml:133]: This of course can be done in another ticket :)

File: tests/run_test_within_docker.sh (hunk 2 above)
Comment (tomuben) [reply tests/run_test_within_docker.sh:25]: No, that not necessary. For "apt install" the -y option is necessary if you install packages in a script, but for update we don't need it.

File: doc/changes/changes_7.1.0.md (hunk 1 above)
Comment (tomuben) [reply doc/changes/changes_7.1.0.md:11]: Done

File: .github/workflows/release_droid_upload_github_release_assets.yml (hunk 6 above)
Comment (tomuben) [reply .github/workflows/release_droid_upload_github_release_assets.yml]: I think that's a great idea.
Although the objective is to provide packages to be used with the devtools::install_url() function, and I believe this function will not check the sha file, it's good to have it if anybody writes it's own install scripts.

I will add this right now to this PR.

File: .github/workflows/release_droid_upload_github_release_assets.yml (hunk 5 above)
Comment (tomuben) [reply .github/workflows/release_droid_upload_github_release_assets.yml]: Hm, this workflow runs only on ubuntu-latest, check line 92

File: .github/workflows/release_droid_upload_github_release_assets.yml (hunk 3 above)
Comment (tomuben) [reply .github/workflows/release_droid_upload_github_release_assets.yml]: No. :)
I removed it.

**Review by morazow — APPROVED** [review]
Looks good!

**Pull request merged**